src/Curriculum/Survey/Survey.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState, FC } from 'react';
+import { useParams } from 'react-router-dom';
+import { useSurveyMeta } from 'src/common/api/curriculum/survey';
+import LoadingContainer from 'src/common/components/LoadingContainer';
+import { SurveyQuestion } from 'src/common/api/curriculum/survey/requests';
+
+import {
+    SimpleGrid,
+    InputGroup,
+    Input,
+    InputRightAddon,
+    Textarea,
+    Slider,
+    SliderTrack,
+    SliderFilledTrack,
+    SliderThumb,
+} from '@chakra-ui/core';
+
+import {
+    SubmitSurvey,
+    SurveyContainer,
+    QuestionLabel,
+  } from './SurveyStyled';
+
+interface Params {
+    surveyId: string;
+}
+
+const answeredIndices = new Set();
+
+const RenderInput: FC<{question:SurveyQuestion, id:number, updateResp:Function}> = ({question, id, updateResp}) => {
```

Comment:
I think you should add a signature to the `updateResp` function. However, it makes line 31 quite long so I would pull those props out into another interface and just use that.

```suggestion
interface RenderInputProps {
  id: number;
  question: SurveyQuestion;
  updateResp: (resp: string, id: number) => any;
}

const RenderInput: FC<RenderInputProps> = ({question, id, updateResp}) => {
```